stdlib/Profile/src/Profile.jl

```diff
@@ -1255,6 +1255,7 @@ function take_heap_snapshot(io::IOStream, all_one::Bool=false)
     Base.@_lock_ios(io, ccall(:jl_gc_take_heap_snapshot, Cvoid, (Ptr{Cvoid}, Cchar), io.handle, Cchar(all_one)))
 end
 function take_heap_snapshot(filepath::String, all_one::Bool=false)
+    mkpath(filepath)
```

Comment:
this seems pretty unexpected for an API to do

Reply:
ok. Removed